clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1181,15 +1181,20 @@ def ShiftOp : CIR_Op<"shift", [Pure]> {
   let summary = "Shift";
   let description = [{
     Shift `left` or `right`, according to the first operand. Second operand is
-    the shift target and the third the amount.
+    the shift target and the third the amount. Second and the thrid operand can
+    be either integer type or vector of integer type. However, they must be
+    either all vector of integer type, or all integer type. If they are vectors,
+    each vector element of the shift target is shifted by the corresponding
+    shift amount in the shift amount vector. 
```

Comment:
Seems like this also matches the more generic shift behavior for GCC extensions, which is great:

https://gcc.gnu.org/onlinedocs/gcc/Vector-Extensions.html
> It is possible to use shifting operators <<, >> on integer-type vectors. The operation is defined as following: {a0, a1, …, an} >> {b0, b1, …, bn} == {a0 >> b0, a1 >> b1, …, an >> bn}. Unlike OpenCL, values of b are not implicitly taken modulo bit width of the base type B, and the behavior is undefined if any bi is greater than or equal to B.

@dkolsen-pgi do you have an opinion/take on this?